tsdb/store.go

```diff
@@ -1180,40 +1180,57 @@ func (s *Store) sketchesForDatabase(dbName string, getSketches func(*Shard) (est
 // Cardinality is calculated exactly by unioning all shards' bitsets of series
 // IDs. The result of this method cannot be combined with any other results.
 //
-func (s *Store) SeriesCardinality(database string) (int64, error) {
+func (s *Store) SeriesCardinality(ctx context.Context, database string) (int64, error) {
 	s.mu.RLock()
 	shards := s.filterShards(byDatabase(database))
 	s.mu.RUnlock()
 
 	var setMu sync.Mutex
 	others := make([]*SeriesIDSet, 0, len(shards))
 
-	s.walkShards(shards, func(sh *Shard) error {
-		index, err := sh.Index()
-		if err != nil {
-			return err
-		}
+	err := s.walkShards(shards, func(sh *Shard) error {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
```

Comment:
nit: you have one pattern like:

```
    select {
    case <-ctx.Done():
      return ctx.Err()
    default:
    }
    // more code
```

And another like:

```
    select {
    case <-ctx.Done():
      return ctx.Err()
    default:
      //more code
    }
```

I prefer the first one. Not blocking.

Reply:
@lesam - Thanks. I like the second pattern, but I used the first to avoid indenting too deeply.  I'll standardize in the future; I think my aesthetics are not yet pure Go.